src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -125,7 +124,7 @@ private bool IsMinimalDelta
         /// NetModules and EnC deltas don't have AssemblyDef record.
         /// We don't emit it for EnC deltas since assembly identity has to be preserved across generations (CLR/debugger get confused otherwise).
         /// </summary>
-        private bool EmitAssemblyDefinition => module.AsAssembly != null && !IsMinimalDelta;
+        private bool EmitAssemblyDefinition => module.OutputKind != OutputKind.NetModule && !IsMinimalDelta;
```

Comment:
We used to sometimes check for ```AsAssembly != null``` and for ```OutputKind != OutputKind.NetModule``` other times. These are equivalent, I normalized all checks to use ```OutputKind```.